src/main/java/nl/novi/serviceconnect/services/QueryService.java

```diff
@@ -0,0 +1,39 @@
+package nl.novi.serviceconnect.services;
+
+import nl.novi.serviceconnect.dtos.CategoryServiceCountDto;
+import nl.novi.serviceconnect.repository.CategoryRepository;
+import nl.novi.serviceconnect.repository.ServiceRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class QueryService implements IQueryService {
```

Comment:
Kleine kanttekening: Ik zou niet de naam "QueryService" gebruiken, maar iets specifieker zijn in wat je query'd (bijvoorbeeld "ServicesQueryService")